### DISCUSSION TITLE
STM32U5 Support Status

### DISCUSSION THREAD

**Discussion by rsgrimes** [hathach/tinyusb#1556] [Q&A] [answered]
According to [git@github.com:hathach/tinyusb.git](git@github.com:hathach/tinyusb.git), the ST STM32U5 is listed as "WIP/partial support", and it uses the "dwc2" driver. I am very interested in using TinyUSB/FreeRTOS on this processor for a rather urgent client need, and would be interested in either helping out or doing it myself, but I certainly don't want to reinvent the wheel. So I have a few "simple" questions:

1. What is the status?
2. If this is in the works, can I get the code as it sits?
3. How can I help?
4. Does it really use the dwc2 driver?

Thanks!

**Comment by cr1901** [comment, marked as answer]
From a quick glance of chapter 63 of [the manual](https://www.st.com/resource/en/reference_manual/rm0456-stm32u575585-armbased-32bit-mcus-stmicroelectronics.pdf): Yes, STM32U5 uses the dwc2 driver. Adding support for STM32U5 means adding various defines to [`dcd_stm32.h`](https://github.com/hathach/tinyusb/blob/master/src/portable/synopsys/dwc2/dwc2_stm32.h).

It's probably easier to test that the port works if you have a dev board, so you can add a directory under [`bsp`](https://github.com/hathach/tinyusb/tree/master/hw/bsp) so you can play with the examples, such as [`cdc_msc`](https://github.com/hathach/tinyusb/tree/master/examples/device/cdc_msc).

**Comment by hathach** [comment]
yeah, most of the newer stm32 is not marked as supported is due to the fact, that I don't have an dev board to actually verify it. If anyone does, please make an PR if possible.

**Reply by Dodotronic** [reply]
I'm trying to implement the audio class on my board but there is some issue, he device is recognized but under "device" I have the name of "MicNode_4_Ch" with a yellow triangle so something is gone wrong after the enumeration.
I'm using STM32U5A5VJTXQ
This micro has high speed USB with integrated PHY
herein attached the source code
[USB_U5.zip](https://github.com/hathach/tinyusb/files/13040355/USB_U5.zip)

**Comment by rsgrimes** [comment]
Will do. I am trying to figure out how to specify which GCC compiler is being used, as I have two on my machine. The build system picks the older compiler that I need for other work that doesn't support cortex M33; I expect the newer will, but I haven't figured out how to tell the build to use that one!

**Comment by hathach** [comment]
supported by #2300
